### PR TITLE
chore: Migrate Python to new directory structure

### PR DESCRIPTION
**Pull request by cclauss**
It would be better if #170, #193, and #195 landed _before_ this change.
```
 14 files changed, 0 insertions(+), 0 deletions(-)
 rename arrays/python/count-inversions.py => algorithms/Python/arrays/count_inversions.py (100%)
 rename {linked-lists/Python => algorithms/Python/linked_lists}/doubly.py (100%)
 rename {linked-lists/Python => algorithms/Python/linked_lists}/singly.py (100%)
 rename {multiplication/python => algorithms/Python/multiplication}/karatsuba.py (100%)
 rename scheduling/python/interval-scheduling.py => algorithms/Python/scheduling/interval_scheduling.py (100%)
 rename searching/python/binary-search.py => algorithms/Python/searching/binary_search.py (100%)
 rename searching/python/jump-search.py => algorithms/Python/searching/jump_search.py (100%)
 rename searching/python/linear-search.py => algorithms/Python/searching/linear_search.py (100%)
 rename sorting/python/bubble-sort.py => algorithms/Python/sorting/bubble_sort.py (100%)
 rename sorting/python/insertion-sort.py => algorithms/Python/sorting/insertion_sort.py (100%)
 rename sorting/python/quick-sort.py => algorithms/Python/sorting/quicksort.py (100%)
 rename sorting/python/selection-sort.py => algorithms/Python/sorting/selection_sort.py (100%)
 rename {strings/python => algorithms/Python/strings}/palindrome.py (100%)
 rename {strings/python => algorithms/Python/strings}/word_count.py (100%)
```

### CODE REVIEW

**Comment by ming-tsai** [comment]
Could you add the Readme file?

File: algorithms/Python/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Python
+
+## Arrays/
```
Comment (ming-tsai) [inline algorithms/Python/README.md:3]: Please remove the `/`

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Could you modify the `workflows/lint_python.yml` to considering only the `.py` and rename its name to `python` extension file?
As this below image
image

**Comment by cclauss** [comment]
No.  That is a bad idea because the performance difference would be negligible yet we would create blind spots.  For example, if we create a `.github/workflows/write_readme.py` script, the tests in that code would never be run.  I find that assumptions about where code will exist in a repo six months in the future are usually wrong.

**Comment by cclauss** [comment]
I put `paths: '**.py'` in #209

**Comment by ming-tsai** [comment]
Nice

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**